include/snowflake/client.h

```diff
@@ -613,6 +632,16 @@ SF_STATUS STDCALL snowflake_get_attribute(
  */
 SF_STMT *STDCALL snowflake_stmt(SF_CONNECT *sf);
 
+/**
+ * Creates sf SNOWFLAKE_STMT context for async queries.
+ *
+ * @param sf The SF_CONNECT context.
+ * @param query_id the query id of the async query.
+ *
+ * @return sfstmt SNOWFLAKE_STMT context for async queries.
+ */
+SF_STMT* STDCALL snowflake_async_stmt(SF_CONNECT *sf, const char *query_id);
```

Comment:
Why we need to create statement specifically for async query execution? I don't see similar approach in JDBC implementation. Adding a new statement attribute would make more sense to me.